code/modules/mob/living/carbon/xenomorph/xeno_defines.dm

```diff
@@ -1035,7 +1037,16 @@
 	return TRUE
 
 /datum/hive_status/proc/update_hugger_limit()
-	playable_hugger_limit = 2 + Ceiling(totalXenos.len / 4)
+	var/iterator = 0
+	var/playable_hugger_limit_counter = 0
+	for(var/mob/living/carbon/xenomorph/cycled_xeno as anything in totalXenos)
+		if(cycled_xeno.counts_for_slots)
+			iterator++
+		if(iterator >= 4)
```

Comment:
This works, but its kinda a weird way to do it. You could save clock cycles by counting up all xenos that count for slots, then divide that count by your divisor (4 here, 3 elsewhere - which should also just be defines so its more obvious that its a tweakable thing), and floor that result (to round down).